examples/Retrier/Client/Program.cs

```diff
@@ -0,0 +1,125 @@
+using Grpc.Net.Client.Configuration;
+using Retry;
+
+namespace Client
+{
+    public class Program
+    {
+        static async Task Main(string[] args)
+        {
+            using var channel = CreateChannel();
+            var client = new Retrier.RetrierClient(channel);
+
+            await UnaryRetry(client);
+
+            Console.WriteLine("Shutting down");
+            Console.WriteLine("Press any key to exit...");
+            Console.ReadKey();
+        }
+
+        private static async Task UnaryRetry(Retrier.RetrierClient client)
+        {
+            Console.WriteLine("Delivering packages...");
+            foreach (var product in Products)
+            {
+                try
+                {
+                    var package = new Package { Name = product };
+                    var call = client.DeliverPackageAsync(package);
+                    var response = await call;
+
+                    #region Print success
+                    Console.ForegroundColor = ConsoleColor.Green;
+                    Console.Write(response.Message);
+                    Console.ResetColor();
+                    Console.Write(" " + await GetRetryCount(call.ResponseHeadersAsync));
+                    Console.WriteLine();
+                    #endregion
+                }
+                catch (RpcException ex)
+                {
+                    #region Print failure
+                    Console.ForegroundColor = ConsoleColor.Red;
+                    Console.WriteLine(ex.Status.Detail);
+                    Console.ResetColor();
+                    #endregion
+                }
+
+                await Task.Delay(TimeSpan.FromSeconds(0.2));
+            }
+        }
+
+        private static GrpcChannel CreateChannel()
+        {
+            var methodConfig = new MethodConfig
+            {
+                Names = { MethodName.Default },
+                RetryPolicy = new RetryPolicy
+                {
+                    MaxAttempts = 10,
+                    InitialBackoff = TimeSpan.FromSeconds(0.5),
+                    MaxBackoff = TimeSpan.FromSeconds(0.5),
+                    BackoffMultiplier = 1,
+                    RetryableStatusCodes = { StatusCode.Unavailable }
+                }
+            };
+
+            return GrpcChannel.ForAddress("http://localhost:5000", new GrpcChannelOptions
+            {
+                ServiceConfig = new ServiceConfig { MethodConfigs = { methodConfig } }
+            });
+        }
+
+        private static async Task<string> GetRetryCount(Task<Metadata> responseHeadersTask)
+        {
+            var headers = await responseHeadersTask;
+            var previousAttemptCount = headers.GetValue("grpc-previous-rpc-attempts");
+            return previousAttemptCount != null ? $"(retry count: {previousAttemptCount})" : string.Empty;
+        }
+
+        private static readonly IList<string> Products = new List<string>
+        {
+            "Secrets of Silicon Valley",
```

Comment:
How did you come up with this list 😄

Reply:
https://github.com/microsoft/sql-server-samples/blob/09f3471b37f082a51a59b0728dfb5768133ace41/samples/databases/northwind-pubs/instpubs.sql#L1693-L1777